x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/action/PutEngineAction.java

```diff
@@ -92,7 +92,7 @@ public String[] indices() {
 
         @Override
         public IndicesRequest indices(String... indices) {
-            return new Request(new Engine(engine.name(), indices, engine.analyticsCollectionName()));
+            return new Request(new Engine(engine.name(), indices, engine.analyticsCollectionName(), System.currentTimeMillis()));
```

Comment:
Do we need the update time to be part of the request? It shouldn't be something that can be specified on the request, so I think it should be internal to the service...

Reply:
@jimczi requested that I have a single constructor on Engine, so I removed the convenience constructor that would not specify the `updated_at` time.

Reply:
Got it! So Engine will need the `updated_at` in the constructor. But, should the `Request` have the `updated_at` information, given that it is never provided as part of the request? I think we could remove that from the request, and specify at Engine creation. WDYT?

Reply:
OK, in order to do this I took `updated_at` out of the Engine constructor and replaced it with a setter, let me know what you think.

Reply:
As discussed offline, we'll leave future updates to `updated_at` to a future PR.